CMSIS/Core_A/Include/cmsis_armcc.h

```diff
@@ -452,7 +452,7 @@ __STATIC_INLINE void __set_FPEXC(uint32_t fpexc)
  */
 
 #define __get_CP(cp, op1, Rt, CRn, CRm, op2) do { register uint32_t tmp __ASM("cp" # cp ":" # op1 ":c" # CRn ":c" # CRm ":" # op2); (Rt) = tmp; } while(0)
```

Comment:
After clarification with compiler experts I would add volatile here as well, just to be sure the actual register access is not optimized.

Reply:
PR updated !